File: be/src/main/java/sg/flow/auth/FlowAccessTokenAuthenticationSuccessHandler.java
package sg.flow.auth;

import java.io.IOException;
import java.util.Map;

import org.springframework.security.core.Authentication;
import org.springframework.security.web.authentication.AuthenticationSuccessHandler;

import com.fasterxml.jackson.databind.ObjectMapper;

import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import sg.flow.models.auth.TokenSet;

public class FlowAccessTokenAuthenticationSuccessHandler implements AuthenticationSuccessHandler {

    @Override
    public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response,
            Authentication authentication) throws IOException, ServletException {
        Object details = authentication.getDetails();
        if (details instanceof TokenSet) {
            String newAccessToken = ((TokenSet) details).getAccessToken();
            String newRefreshToken = ((TokenSet) details).getRefreshToken();

            // Build JSON or any response format
            response.setContentType("application/json");
            response.setCharacterEncoding("UTF-8");
            String json = new ObjectMapper().writeValueAsString(Map.of(
                    "accessToken", newAccessToken,
                    "refreshToken", newRefreshToken));
            response.getWriter().write(json);
            response.getWriter().flush();
        }
    }

}

File: be/src/main/kotlin/sg/flow/auth/FlowAccessTokenAuthenticationSuccessHandler.kt
package sg.flow.auth

import com.fasterxml.jackson.databind.ObjectMapper
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.core.Authentication
import org.springframework.security.web.authentication.AuthenticationSuccessHandler
import sg.flow.models.auth.TokenSet
import java.io.IOException

class FlowAccessTokenAuthenticationSuccessHandler : AuthenticationSuccessHandler {

    @Throws(IOException::class)
    override fun onAuthenticationSuccess(
        request: HttpServletRequest,
        response: HttpServletResponse,
        authentication: Authentication,
    ) {
        val details = authentication.details
        if (details is TokenSet) {
            response.contentType = "application/json"
            response.characterEncoding = "UTF-8"
            val json = ObjectMapper().writeValueAsString(
                mapOf(
                    "accessToken" to details.accessToken,
                    "refreshToken" to details.refreshToken
                )
            )
            response.writer.write(json)
            response.writer.flush()
        }
    }
} 